wp-content/themes/core/components/navigation/Navigation_Controller.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Templates\Components\navigation;
+
+use Tribe\Libs\Utils\Markup_Utils;
+use Tribe\Project\Nav_Menus\Menu;
+use Tribe\Project\Templates\Components\Abstract_Controller;
+
+class Navigation_Controller extends Abstract_Controller {
+
+	public const LOCATION         = 'location';
+	public const CLASSES          = 'classes';
+	public const ATTRS            = 'attrs';
+	public const NAV_LIST_CLASSES = 'nav_list_classes';
+	public const NAV_MENU_ARGS    = 'nav_menu_args';
+
+	private const DEFAULT_NAV_MENU_ARGS = [
+		'container'   => false,
+		'menu_class'  => '',
+		'menu_id'     => '',
+		'depth'       => 1,
+		'items_wrap'  => '%3$s',
+		'fallback_cb' => false,
+	];
+
+	private string $location;
+	private array $classes;
+	private array $attrs;
+	private array $nav_list_classes;
+	private array $nav_menu_args;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->location         = (string) $args[ self::LOCATION ];
+		$this->classes          = (array) $args[ self::CLASSES ];
+		$this->attrs            = (array) $args[ self::ATTRS ];
+		$this->nav_list_classes = (array) $args[ self::NAV_LIST_CLASSES ];
+		$this->nav_menu_args    = $this->parse_menu_args( $args[ self::NAV_MENU_ARGS ] );
+	}
+
+	protected function defaults(): array {
```

Comment:
This will be fixed automatically with `phpcbf` in the v2 rules, but class order should go public methods, protected and then private.